### PR TITLE
saving reset pin

### PR DESCRIPTION
**Pull request by universam1**
should cover #56

Reset could be wired to ESP8266 reset pin and get resetted at boot.

### CODE REVIEW

**Comment by universam1** [comment]
This PR @sandeepmistry has been tested with ESP8266 and AI-Thinker RA-01 and works perfectly!

File: src/LoRa.cpp [hunk 1]
```diff
@@ -66,17 +66,20 @@ int LoRaClass::begin(long frequency)
...
-  digitalWrite(_reset, LOW);
-  delay(10);
-  digitalWrite(_reset, HIGH);
-  delay(10);
-
   // set SS high
   digitalWrite(_ss, HIGH);
 
+  if (_reset != -1)
+  {
```
Comment (sandeepmistry) [inline src/LoRa.cpp]: Could you please make this bracket consistent with the rest of the library.

File: src/LoRa.cpp [hunk 2]
```diff
@@ -502,7 +505,6 @@ void LoRaClass::writeRegister(uint8_t address, uint8_t value)
 uint8_t LoRaClass::singleTransfer(uint8_t address, uint8_t value)
 {
   uint8_t response;
-
```
Comment (sandeepmistry) [inline src/LoRa.cpp]: Could you please undo this change?

File: API.md [hunk 3]
```diff
@@ -32,6 +32,13 @@ LoRa.setPins(ss, reset, dio0);
 
 This call is optional and only needs to be used if you need to change the default pins used.
 
+### Saving pin (tested on ESP8266)
```
Comment (sandeepmistry) [inline API.md]: Even though it's only been tested on ESP8266, I think we can leave out the text in the brackets.

Comment (universam1) [reply API.md]: fine with me

File: src/LoRa.cpp (hunk 1 above)
Comment (universam1) [reply src/LoRa.cpp]: np - my stylecheck is really enemy with your file... I dont mind.

File: src/LoRa.cpp (hunk 2 above)
Comment (universam1) [reply src/LoRa.cpp]: sure

**Comment by universam1** [comment]
requests applied

**Comment by sandeepmistry** [comment]
@universam1 thanks!
